stonybook/pipeline/quote/quote.py

```diff
@@ -0,0 +1,961 @@
+        conversations = []
+        currentConversation = []
+
+        for i, quoteObj in enumerate(self.quotes):
+            if len(currentConversation) == 0:
+                currentConversation.append(quoteObj)
+            else:
+                previousSentenceId = int(currentConversation[-1].endSentId)
+                if int(quoteObj.startSentId) - previousSentenceId <= 3:
+                    currentConversation.append(quoteObj)
+                else:
+                    conversations.append(currentConversation)
+                    currentConversation = [quoteObj]
+
+        self.conversations = conversations
+
+    def searchUpwards(self, conversationIdx, currentIdx):
+        i = currentIdx - 2
+        while (i >= 0):
+            if self.conversations[conversationIdx][i].mention != None:
+                self.conversations[conversationIdx][currentIdx].mention = self.conversations[conversationIdx][i].mention
+                self.conversations[conversationIdx][currentIdx].mention_start = self.conversations[conversationIdx][
+                    i].mention_start
+                self.conversations[conversationIdx][currentIdx].mention_end = self.conversations[conversationIdx][
+                    i].mention_end
+                return True
+            i = i - 2
+        return False
+
+    def searchDownwards(self, conversationIdx, currentIdx):
+        i = currentIdx + 2
+        while (i < len(self.conversations[conversationIdx])):
+            if self.conversations[conversationIdx][i].mention != None:
+                self.conversations[conversationIdx][currentIdx].mention = self.conversations[conversationIdx][i].mention
+                self.conversations[conversationIdx][currentIdx].mention_start = self.conversations[conversationIdx][
+                    i].mention_start
+                self.conversations[conversationIdx][currentIdx].mention_end = self.conversations[conversationIdx][
+                    i].mention_end
+                return True
+            i = i + 2
+        return False
+
+    def correctMentionsInConversations(self):
+        for i, conversation in enumerate(self.conversations):
+            for j, quoteObj in enumerate(conversation):
+                if quoteObj.mention == None:
+                    status = self.searchUpwards(i, j)
+                    if status == False:
+                        self.searchDownwards(i, j)
+        self.updateQuotes()
+
+    def updateQuotes(self):
+        for conversation in self.conversations:
+            for quoteObj in conversation:
+                self.quotes[quoteObj.quoteListIdx].mention = quoteObj.mention
+                self.quotes[quoteObj.quoteListIdx].mention_start = quoteObj.mention_start
+                self.quotes[quoteObj.quoteListIdx].mention_end = quoteObj.mention_end
+
+    def populateQuoteListIdx(self):
+        for i, quoteObj in enumerate(self.quotes):
+            self.quotes[i].quoteListIdx = i
+
+    def looseConversation(self):
```

Comment:
This whole function can be reduced to just the loop in line 905 and the if condition in line 906 can just check if the mention is None, the index is greater than 0, and the previous mention is not None. Then, you can just assign the previous mention to the current quote.